File: src/modules/models/BookingModel.js
class BookingModel {
    constructor() {
        this.bookings = [];
    }

    addBooking(booking) {
        this.bookings.push(booking);
    }

    isDuplicate(userId, flightId, seatNumbers) {
        return this.bookings.some(b =>
            b.userId === userId &&
            b.flightId === flightId &&
            seatNumbers.some(seat => b.seats.includes(seat))
        );
    }

    getAllBookings() {
        return this.bookings;
    }
}

module.exports = BookingModel;

File: src/modules/Booking&Seat Module/model/BookingModel.java

package model;

import java.util.*;

public class BookingModel {
    private List<Booking> bookings = new ArrayList<>();

    public void addBooking(Booking booking){
        bookings.add(booking);
    }

    public boolean isDuplicate(String userId, String flightId, List<String> seats){
        for(Booking b : bookings){
            if(b.userId.equals(userId) && b.flightId.equals(flightId)){
                for(String s : seats){
                    if(b.seats.contains(s)) return true;
                }
            }
        }
        return false;
    }

    public List<Booking> getAllBookings(){
        return bookings;
    }
}
